diff --git a/modules/mailchimp_signup/mailchimp_signup.module b/modules/mailchimp_signup/mailchimp_signup.module
index ff93730..e663b81 100644
--- a/modules/mailchimp_signup/mailchimp_signup.module
+++ b/modules/mailchimp_signup/mailchimp_signup.module
@@ -390,7 +390,7 @@ function mailchimp_signup_subscribe_form_submit($form, &$form_state) {
     $result = mailchimp_subscribe($list_id, $email, $mergevars, $interests, $signup->settings['doublein']);
     // Let other modules act on the results in hook_form_alter.
     $form_state['mailchimp_results'] = $result;
-    if (empty($result) || ($result->success === FALSE)) {
+    if (empty($result) || (isset($result->success) && $result->success === FALSE)) {
       drupal_set_message(t('There was a problem with your newsletter signup to %list.', array(
         '%list' => $list_details[$list_id]->name,
       )), 'warning');
